robot-command/src/main/java/org/obolibrary/robot/ExplainCommand.java

```diff
@@ -0,0 +1,146 @@
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLAnnotationAssertionAxiom;
+import org.semanticweb.owlapi.model.OWLAxiom;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
+import org.semanticweb.owlapi.util.SimpleShortFormProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles inputs and outputs for the {@link ExplainOperation}.
+ *
+ * @author <a href="mailto:balhoff@renci.org">Jim Balhoff</a>
+ */
+public class ExplainCommand implements Command {
+  /** Logger. */
+  private static final Logger logger = LoggerFactory.getLogger(ExplainCommand.class);
+
+  /** Store the command-line options for the command. */
+  private Options options;
+
+  public ExplainCommand() {
+    Options o = CommandLineHelper.getCommonOptions();
+    o.addOption("i", "input", true, "load ontology from a file");
+    o.addOption("I", "input-iri", true, "load ontology from an IRI");
+    o.addOption("r", "reasoner", true, "reasoner to use: ELK, HermiT, JFact");
+    o.addOption("a", "axiom", true, "the axiom to explain");
+    o.addOption("m", "max", true, "the maximum number of explanations to retrieve");
+    o.addOption("f", "format", true, "the explanation result format: MD, OFN, etc.");
+    o.addOption("o", "output", true, "save explanation to a file");
+    options = o;
+  }
+
+  @Override
+  public String getName() {
+    return "explain";
+  }
+
+  @Override
+  public String getDescription() {
+    return "explain derivation of an inferred axiom";
+  }
+
+  @Override
+  public String getUsage() {
+    return "robot explain --input <file> --reasoner <reasoner> --axiom <axiom> <output>";
+  }
+
+  @Override
+  public Options getOptions() {
+    return options;
+  }
+
+  /**
+   * Handle the command-line and file operations for the ExplainOperation.
+   *
+   * @param args strings to use as arguments
+   */
+  @Override
+  public void main(String[] args) {
+    try {
+      execute(null, args);
+    } catch (Exception e) {
+      CommandLineHelper.handleException(e);
+    }
+  }
+
+  @Override
+  public CommandState execute(CommandState state, String[] args) throws Exception {
+    CommandLine line = CommandLineHelper.getCommandLine(getUsage(), getOptions(), args);
+    if (line == null) {
+      return null;
+    }
+    if (state == null) {
+      state = new CommandState();
+    }
+    IOHelper ioHelper = CommandLineHelper.getIOHelper(line);
+    state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
+    OWLOntology ontology = state.getOntology();
+    QuotedEntityChecker checker = new QuotedEntityChecker();
+    checker.setIOHelper(ioHelper);
+    checker.addProvider(new SimpleShortFormProvider());
+    checker.addProperty(OWLManager.getOWLDataFactory().getRDFSLabel());
+    if (ontology != null) {
+      checker.addAll(ontology);
+    }
+    int max = 1;
```

Comment:
This could be a bit more robust for catching errors. We also have a `getDefaultValue` method.
```
String maxString = CommandLineHelper.getDefaultValue(line, "max", "1");
int max;
try {
  max = Integer.parseInt(maxString);
} catch (NumberFormatException e) {
  throw new Exception(String.format(maxTypeError, maxString));
}
```

Where the error messages at the top are something like...
```
private static final String NS = "explain#";
private static final String maxTypeError = NS + "MAX TYPE ERROR --max ('%s') must be an integer"
```

This allows our `ExceptionHelper` to parse the error message and display a nice output in the console. See error message examples in our current docs: http://robot.obolibrary.org/convert#check-arg-error